packages/nimble-components/src/wafer-map/specs/features/rendering.md

```diff
@@ -0,0 +1,189 @@
+| 2 thread | update     | main         | 4.0ms     | 4.1ms     | 4.2ms     | 1M dies |
+|          | renderDies | worker       | 204ms     | 207ms     | 209ms     | 1M dies |
+
+#### Size Limitations
+
+Because we are reaching out to support very large amounts of data, an issue that is popping out is the limits of the component's canvas size to the amount of displayed data. For example we could find ourselves in the situation to render more dies than there are pixels on the canvas, at least in the zoomed out state.
+
+The temporary solution to this issue is creating a new validity warning or event which will signal this state to the user when the count of individual dies will overtake the size of the canvas. A heuristic which will calculate this based on the zoom level and the component height and width will be implemented and used during data validation.
+
+The warning will be displayed by the existing warning signaling mechanism used in the Ozone application.
+![warning display](resources/warning-integration.png)
+
+### Performance Testing
+
+We wish to measure functions performance of wafer map using the existing lighthouse setup paired with windows.performance tools.
+
+Example:
+
+```TS
+    const functionName = 'createWaferMap';
+    performance.mark(`${functionName}_start`);
+    const wafermap = document.createElement('nimble-wafer-map');
+    document.body.appendChild(wafermap);
+    performance.mark(`${functionName}_end`);
+    performance.measure(functionName, `${functionName}_start`, `${functionName}_end`);
+```
+
+The example shows how custom markers can be set in the tested code. The result of these user marks can be observed inside lighthouse like in the screenshot bellow.
+
+![User Timing and measures](./resources/user-timing-marks-and-measures.PNG)
+
+We wish to extend the existing lighthouse CI testing to include the wafermap component and record various metrics about the performance it has when rendering multiple generated data loads. They will include data ingestion, rendering time, rerendering when user events get triggered and data changes.
+
+These records will be made in a separate web page from the existing performance tests to avoid polluting the results. They will not be preserved over time, each CI record being available for download after build.
+
+Our target goals are:
+
+1. For wafers >200k and <1m die
+
+    1. Initial rendering: <1 second
+    1. Zoom/pan actions: <500ms
+    1. Configuration changes that don't modify the dataset (e.g. show labels):
+       <500ms
+    1. Configuration changes that modify the dataset (e.g. replace layers): <1 second
+
+1. For wafers >1m and <10m die
+    1. Initial rendering: <10 seconds
+    1. Zoom/pan actions: <1 second
+    1. Interactions that don't modify the dataset (e.g. show labels): <1 second
+    1. Interactions that modify the dataset (e.g. replace layers): <10 seconds
+
+## Alternative Implementations / Designs
+
+### Alternative Data Structures and Interfaces
```

Comment:
When listing alternatives it's helpful to include the rationale for not choosing that alternative.